doc/v5.0/man3/MPI_Type_lb.3
.\" -*- nroff -*-
.\" Copyright 2010 Cisco Systems, Inc.  All rights reserved.
.\" Copyright 2006-2008 Sun Microsystems, Inc.
.\" Copyright (c) 1996 Thinking Machines Corporation
.\" Copyright (c) 2020      Google, LLC. All rights reserved.
.\" $COPYRIGHT$
.TH MPI_Type_lb 3 "Sep 30, 2021" "5.0.0rc1" "Open MPI"
.SH NAME
\fBMPI_Type_lb\fP \- Returns the lower bound of a data type -- use of this routine is deprecated.

.SH SYNTAX
.ft R
.SH C Syntax
.nf
#include <mpi.h>
int MPI_Type_lb(MPI_Datatype \fIdatatype\fP, MPI_Aint\fI *displacement\fP)

.fi
.SH Fortran Syntax
.nf
INCLUDE 'mpif.h'
MPI_TYPE_LB(\fIDATATYPE, DISPLACEMENT, IERROR\fP)
	INTEGER	\fIDATATYPE, DISPLACEMENT, IERROR\fP


.fi
.SH INPUT PARAMETER
.ft R
.TP 1i
datatype
Datatype (handle).
.sp
.SH OUTPUT PARAMETERS
.ft R
.TP 1i
displacement
Displacement of lower bound from origin, in bytes (integer).
.ft R
.TP 1i
IERROR
Fortran only: Error status (integer).

.SH DESCRIPTION
.ft R
Note that use of this routine is \fIdeprecated\fP as of MPI-2. Please use MPI_Type_get_extent instead.
.sp
MPI_Type_lb returns the lower bound of a data type. This may differ from zero if the type was constructed using MPI_LB.
.sp
The "pseudo-datatypes," MPI_LB and MPI_UB, can be used, respectively, to mark the lower bound (or the upper bound) of a datatype. These pseudo-datatypes occupy no space (extent (MPI_LB) = extent (MPI_UB) =0. They do not affect the size or count of a datatype, and do not affect the context of a message created with this datatype. However, they do affect the definition of the extent of a datatype and, therefore, affect the outcome of a replication of this datatype by a datatype constructor.
.sp
In general, if
.sp
.nf
    Typemap = {(type0, disp0), ..., (type(n-1), disp(n-1)}
.fi
.sp
then the lower bound of Typemap is defined to be
.nf

                  (min(j) disp(j)                          if no entry has
    lb(Typemap) = (                                        basic type lb
                  (min(j) {disp(j) such that type(j) = lb} otherwise

.fi
Similarly, the upper bound of Typemap is defined to be
.nf

                  (max(j) disp(j) + sizeof((type(j)) + e   if no entry has
    ub(Typemap) = (                                        basic type ub
                  (max(j) {disp(j) such that type(j) = ub} otherwise

Then

    extent(Typemap) = ub(Typemap) - lb(Typemap)
.fi
.sp
If type(i) requires alignment to a byte address that is a multiple of k(i),
then e is the least nonnegative increment needed to round extent(Typemap) to the next multiple of max(i) k(i).

.SH ERRORS
Almost all MPI routines return an error value; C routines as the value of the function and Fortran routines in the last argument.
.sp
Before the error value is returned, the current MPI error handler is
called. By default, this error handler aborts the MPI job, except for I/O function errors. The error handler may be changed with MPI_Comm_set_errhandler; the predefined error handler MPI_ERRORS_RETURN may be used to cause error values to be returned. Note that MPI does not guarantee that an MPI program can continue past an error.

.SH SEE ALSO
.ft R
.sp
MPI_Type_get_extent
.br



doc/v1.4/man3/MPI_Type_lb.3.php
<?php
$topdir = "../../..";
$title = "MPI_Type_lb(3) man page (version 1.4.5)";
$meta_desc = "Open MPI v1.4.5 man page: MPI_TYPE_LB(3)";

include_once("$topdir/doc/nav.inc");
include_once("$topdir/includes/header.inc");
?>
<p> <a href="../">&laquo; Return to documentation listing</a></p>
  <!-- manual page source format generated by PolyglotMan v3.2, -->
<!-- available at http://polyglotman.sourceforge.net/ -->

<body bgcolor='white'>
<a href='#toc'>Table of Contents</a><p>

<h2><a name='sect0' href='#toc0'>Name</a></h2>
<b>MPI_Type_lb</b> - Returns the lower bound of a data type. This may differ
from zero if the type was constructed using MPI_LB. Use of this routine
is deprecated.
<p>
<h2><a name='sect1' href='#toc1'>Syntax</a></h2>

<h2><a name='sect2' href='#toc2'>C Syntax</a></h2>
<br>
<pre>#include &lt;mpi.h&gt;
int MPI_Type_lb(MPI_Datatype datatype, MPI_Aint *displacement)
</pre>
<h2><a name='sect3' href='#toc3'>Fortran Syntax</a></h2>
<br>
<pre>INCLUDE &rsquo;mpif.h&rsquo;
MPI_TYPE_LB(DATATYPE, DISPLACEMENT, IERROR)
<tt> </tt>&nbsp;<tt> </tt>&nbsp;INTEGER<tt> </tt>&nbsp;<tt> </tt>&nbsp;DATATYPE, DISPLACEMENT, IERROR
</pre>
<h2><a name='sect4' href='#toc4'>Input Parameter</a></h2>

<dl>

<dt>datatype      </dt>
<dd>Datatype (handle). <p>
</dd>
</dl>

<h2><a name='sect5' href='#toc5'>Output Parameters</a></h2>

<dl>

<dt>displacement
      </dt>
<dd>Displacement of lower bound from origin, in bytes (integer). </dd>

<dt>IERROR
</dt>
<dd>Fortran only: Error status (integer).
<p> </dd>
</dl>

<h2><a name='sect6' href='#toc6'>Description</a></h2>
Note that use of this
routine is <i>deprecated</i> as of MPI-2. Please use <a href="../man3/MPI_Type_get_extent.3.php">MPI_Type_get_extent</a> instead.
 <p>
This deprecated routine is not available in C++.  <p>
MPI_Type_lb returns the
lower bound of a data type.  <p>
The "pseudo-datatypes," MPI_LB and MPI_UB, can
be used, respectively, to mark the lower bound (or the upper bound) of
a datatype. These pseudo-datatypes occupy no space (extent (MPI_LB) = extent
(MPI_UB) =0. They do not affect the size or count of a datatype, and do
not affect the context of a message created with this datatype. However,
they do affect the definition of the extent of a datatype and, therefore,
affect the outcome of a replication of this datatype by a datatype constructor.
<p>
In general, if <p>
<br>
<pre>    Typemap = {(type0, disp0), ..., (type(n-1), disp(n-1)}
</pre><p>
then the lower bound of Typemap is defined to be  <br>
<pre>                  (min(j) disp(j)                          if no entry
has
    lb(Typemap) = (                                        basic type lb
                  (min(j) {disp(j) such that type(j) = lb} otherwise
</pre>Similarly, the upper bound of Typemap is defined to be <br>
<pre>                  (max(j) disp(j) + sizeof((type(j)) + e   if no entry
has
    ub(Typemap) = (                                        basic type ub
                  (max(j) {disp(j) such that type(j) = ub} otherwise
Then
    extent(Typemap) = ub(Typemap) - lb(Typemap)
</pre><p>
If type(i) requires alignment to a byte address that is a multiple of k(i),
then e is the least nonnegative increment needed to round extent(Typemap)
to the next multiple of max(i) k(i).
<p>
<h2><a name='sect7' href='#toc7'>Errors</a></h2>
Almost all MPI routines return
an error value; C routines as the value of the function and Fortran routines
in the last argument. C++ functions do not return errors. If the default
error handler is set to MPI::ERRORS_THROW_EXCEPTIONS, then on error the
C++ exception mechanism will be used to throw an MPI:Exception object. <p>
Before
the error value is returned, the current MPI error handler is called. By
default, this error handler aborts the MPI job, except for I/O function
errors. The error handler may be changed with <a href="../man3/MPI_Comm_set_errhandler.3.php">MPI_Comm_set_errhandler</a>; the
predefined error handler MPI_ERRORS_RETURN may be used to cause error values
to be returned. Note that MPI does not guarantee that an MPI program can
continue past an error.
<p>
<h2><a name='sect8' href='#toc8'>See Also</a></h2>
<p>
<a href="../man3/MPI_Type_get_extent.3.php">MPI_Type_get_extent</a> <br>

<p> <p>

<hr><p>
<a name='toc'><b>Table of Contents</b></a><p>
<ul>
<li><a name='toc0' href='#sect0'>Name</a></li>
<li><a name='toc1' href='#sect1'>Syntax</a></li>
<li><a name='toc2' href='#sect2'>C Syntax</a></li>
<li><a name='toc3' href='#sect3'>Fortran Syntax</a></li>
<li><a name='toc4' href='#sect4'>Input Parameter</a></li>
<li><a name='toc5' href='#sect5'>Output Parameters</a></li>
<li><a name='toc6' href='#sect6'>Description</a></li>
<li><a name='toc7' href='#sect7'>Errors</a></li>
<li><a name='toc8' href='#sect8'>See Also</a></li>
</ul>


<p> <a href="../">&laquo; Return to documentation listing</a></p>
<?php
include_once("$topdir/includes/footer.inc");
